### PR TITLE
Add Auth Rest API implementation

### PR DESCRIPTION
**Pull request by Ninjeneer**

### CODE REVIEW

File: src/provider/CentralServerProvider.tsx [hunk 1]
```diff
@@ -418,7 +418,7 @@ export default class CentralServerProvider {
     const tenant = await this.getTenant(tenantSubDomain);
     // Call
     const result = await this.axiosInstance.get(
-      `${this.buildCentralRestServerServiceAuthURL(tenant)}/${ServerAction.VERIFY_EMAIL}`, {
+      `${this.buildRestServerAuthURL(tenant)}/${ServerAction.VERIFY_EMAIL}`, {
```
Comment (jerome-benoit) [inline src/provider/CentralServerProvider.tsx:421]: Same comment than the one done on the dashboard.

File: src/provider/CentralServerProvider.tsx [hunk 2]
```diff
@@ -325,7 +325,7 @@ export default class CentralServerProvider {
     // Get the Tenant
     const tenant = await this.getTenant(params.tenantSubDomain);
     // Call
-    const result = await this.axiosInstance.get(`${this.buildCentralRestServerServiceAuthURL(tenant)}/${ServerAction.CHECK_END_USER_LICENSE_AGREEMENT}`, {
+    const result = await this.axiosInstance.get(`${this.buildRestServerAuthURL(tenant)}/${ServerAction.CHECK_END_USER_LICENSE_AGREEMENT}`, {
```
Comment (jerome-benoit) [inline src/provider/CentralServerProvider.tsx:328]: Same comment than the one done on the dashboard. 

Please test your changes ;)

**Review by jerome-benoit — CHANGES_REQUESTED** [review]

File: src/provider/CentralServerProvider.tsx [hunk 3]
```diff
@@ -380,7 +380,7 @@ export default class CentralServerProvider {
     const tenant = await this.getTenant(tenantSubDomain);
     // Call
     const result = await this.axiosInstance.post(
-      `${this.buildCentralRestServerServiceAuthURL(tenant)}/${ServerAction.PASSWORD_RESET}`,
+      `${this.buildRestServerAuthURL(tenant)}/password/${ServerAction.REST_RESET}`,
```
Comment (LucasBrazi06) [inline src/provider/CentralServerProvider.tsx:383]: Use REST_PASSWORD_REST with full path and not REST_RESET which means nothing when you read the code!

File: src/provider/CentralServerProvider.tsx [hunk 4]
```diff
@@ -399,7 +399,7 @@ export default class CentralServerProvider {
...
```
Comment (LucasBrazi06) [inline src/provider/CentralServerProvider.tsx:402]: Use full path

File: src/provider/CentralServerProvider.tsx [hunk 5]
```diff
@@ -418,7 +418,7 @@ export default class CentralServerProvider {
     const tenant = await this.getTenant(tenantSubDomain);
     // Call
     const result = await this.axiosInstance.get(
-      `${this.buildCentralRestServerServiceAuthURL(tenant)}/${ServerAction.VERIFY_EMAIL}`, {
+      `${this.buildRestServerAuthURL(tenant)}/mail/${ServerAction.REST_CHECK}`, {
```
Comment (LucasBrazi06) [inline src/provider/CentralServerProvider.tsx:421]: Use full path

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]

File: src/types/Server.tsx [hunk 6]
```diff
@@ -362,4 +369,16 @@ export enum ServerAction {
...
   HTTP_RESPONSE = 'HttpResponse',
   HTTP_ERROR = 'HttpError',
+
+  // RESTful API
+  REST_SIGNIN = 'signin',
+  REST_SIGNON = 'signon',
+  REST_SIGNOUT = 'signout',
+  REST_END_USER_LICENSE_AGREEMENT = 'eula',
+  REST_END_USER_LICENSE_AGREEMENT_CHECK = 'eula/check',
+  REST_CHECK = 'check',
```
[the remark this answers is not in the document]
Comment (Ninjeneer) [reply src/types/Server.tsx:379]: This naming has been asked by Jérôme, cf : https://github.com/LucasBrazi06/ev-server/pull/1865

**Review by jerome-benoit — APPROVED** [review]

**Review by jerome-benoit — CHANGES_REQUESTED** [review]
+ Changes requested by Serge not made.
+ Conflicts to resolve.

**Review by jerome-benoit — APPROVED** [review]

**Pull request merged**